### ISSUE TITLE
Capacitor App Plugin - Events not firing - appUrlOpen / getLaunchUrl()

### ISSUE THREAD

**Issue by polisettyManoj** [capacitor-community/electron#194]
I'm using the capacitor App plugin and trying to get URL from which the app is opened (I need to handle deep link)

npm install @capacitor/app
npx cap sync

App.addListener('appUrlOpen', data => {
  console.log('App opened with URL:', data);   //  **not triggerring**
});

const checkAppLaunchUrl = async () => {
  const { url } = await App.getLaunchUrl();

  console.log('App opened with URL: ' + url); //  **url always empty**
};

appUrlOpen event needs to fire and getLaunchUrl need to give deeplink Url

**Desktop - Windows

@IT-MikeS  I appreciate your quick response

**Comment by IT-MikeS** [comment]
Have you followed the guide in the docs.

https://capacitor-community.github.io/electron/docs/deeplinking

**Comment by polisettyManoj** [comment]
@IT-MikeS yes followed.

**Comment by polisettyManoj** [comment]
@IT-MikeS I created a sample application by following these steps

1. ionic start appElectron blank --type=angular --capacitor
2. npm i @capacitor-community/electron
3. ionic build
4. npx cap add @capacitor-community/electron


after that added capacitor/app plugin and followed the steps given in this document - https://capacitor-community.github.io/electron/docs/deeplinking


npm install @capacitor/app
npx cap sync

App.addListener('appUrlOpen', data => {
console.log('App opened with URL:', data); // not triggerring
});

const checkAppLaunchUrl = async () => {
const { url } = await App.getLaunchUrl();

console.log('App opened with URL: ' + url); // url always empty
};

**Comment by IT-MikeS** [comment]
> @IT-MikeS I created a sample application by following these steps
> 
> 1. ionic start appElectron blank --type=angular --capacitor
> 2. npm i @capacitor-community/electron
> 3. ionic build
> 4. npx cap add @capacitor-community/electron
> 
> after that added capacitor/app plugin and followed the steps given in this document - https://capacitor-community.github.io/electron/docs/deeplinking
> 
> npm install @capacitor/app 
> npx cap sync
> 


Did you cap sync with the electron command? (npx cap sync @capacitor-community/electron)

**Comment by polisettyManoj** [comment]
@IT-MikeS yes (npx cap sync @capacitor-community/electron) this is also done

**Comment by polisettyManoj** [comment]
@IT-MikeS  I'm using "electron:build-windows": "npm run build && electron-builder build --windows --publish=never" command to build the application and tested. getLaunchURL is still empty

**Comment by polisettyManoj** [comment]
> I'm using the capacitor App plugin and trying to get URL from which the app is opened (I need to handle deep link)
> 
> npm install @capacitor/app npx cap sync
> 
> App.addListener('appUrlOpen', data => { console.log('App opened with URL:', data); // **not triggerring** });
> 
> const checkAppLaunchUrl = async () => { const { url } = await App.getLaunchUrl();
> 
> console.log('App opened with URL: ' + url); // **url always empty** };
> 
> appUrlOpen event needs to fire and getLaunchUrl need to give deeplink Url
> 
> **Desktop - Windows
> 
> @IT-MikeS I appreciate your quick response

@IT-MikeS if possible you can create a sample project by following these steps, it won't take much time. can you test it in your Machine ?

**Comment by polisettyManoj** [comment]
@IT-MikeS I went through capacitor/app V3 plugin code (https://github.com/ionic-team/capacitor-plugins/blob/main/app/src/web.ts) they didn't implement several methods for the web. getLaunchUrl is simply returning an empty string.

![image](https://user-images.githubusercontent.com/22149014/176192292-9d028e9e-049c-4679-84e6-4e97d946cd96.png)

**Comment by polisettyManoj** [comment]
@IT-MikeS any update on this? are you looking into this issue ?

**Comment by IT-MikeS** [comment]
It's on my radar but it will be a while before I dig in due to other commitments and life balance. However PRs are always welcome should you feel inclined to investigate and fix the issue in your own time.

**Comment by polisettyManoj** [comment]
@IT-MikeS I tried but was unable to fix it. can you fix it in your own time? or guide me to fix the issue.

--> I need to change capacitor/app plugin code ?

**Comment by polisettyManoj** [comment]
Okay @IT-MikeS :)

**Comment by polisettyManoj** [comment]
@mlynch @danielsogl @vevedh @leMaik @stewones @coreyjv @jgoux @MathisTLD @challenger71498 @jdgjsag67251 can anyone help on this ?

**Comment by jdgjsag67251** [comment]
AFAIK there are no official Capacitor plugins that support Electron. I have made two (https://github.com/ionic-team/capacitor-plugins/pull/792 & https://github.com/ionic-team/capacitor-plugins/pull/793), but none of them have seen any action.

**Comment by polisettyManoj** [comment]
@jdgjsag67251 how to achieve this? any guidance?

**Comment by JuliusSkrisa** [comment]
@polisettyManoj if the app "App.addListener('appUrlOpen'...." doesnt work for you. you can expose function from main process that will return the url. 

In preload.ts expose the function ipcRenderer.on (not secure) or create channel between main and renderer process to pass the URL information. In main process you can get it by calling 
`ipcRenderer.on('appUrlOpen', (url) => { console.log(url)})`

you can read this to understand how to create a channel https://www.debugandrelease.com/the-ultimate-electron-guide/

**Comment by polisettyManoj** [comment]
I will try @JuliusSkrisa. Thank you very much
